src/bootstrap/compile.rs

```diff
@@ -70,6 +70,33 @@ impl Step for Std {
 
         builder.ensure(StartupObjects { compiler, target });
```

Comment:
I think this should happen after the `if` below to remove the conditional inside of `StartupObjects`

Reply:
I moved it below and also added an assert in `StartupObjects`.